src/app/cluster/duck/sagas.ts

```diff
@@ -0,0 +1,260 @@
+import { takeLatest, select, race, take, call, put, delay, } from 'redux-saga/effects';
+import { ClientFactory } from '../../../client/client_factory';
+import { IClusterClient } from '../../../client/client';
+import { MigResource, MigResourceKind } from '../../../client/resources';
+import {
+  ClusterRegistryResource,
+  ClusterRegistryResourceKind,
+  CoreNamespacedResource,
+  CoreNamespacedResourceKind,
+} from '../../../client/resources';
+import {
+  createClusterRegistryObj,
+  createTokenSecret,
+  createMigCluster,
+  updateClusterRegistryObj,
+  updateTokenSecret,
+} from '../../../client/resources/conversions';
+
+import { Creators } from './actions';
+import { alertErrorTimeout } from '../../common/duck/actions';
+import {
+  createAddEditStatus,
+  AddEditState,
+  AddEditMode,
+  IAddEditStatus,
+  AddEditWatchTimeout,
+  AddEditWatchTimeoutPollInterval,
+  AddEditConditionCritical,
+  createAddEditStatusWithMeta,
+  AddEditConditionReady,
+  AddEditDebounceWait,
+} from '../../common/add_edit_state';
+
+function* addClusterRequest(action)  {
+  // TODO: Need to improve this to fall into the failed create state with rollback
```

Comment:
add ui state for isLoading as soon as this is dispatched.

Reply:
seeing the add fail with "clusterRef must reference a cluster" but shows as connected when I refresh the UI